Flavours/stellio-flavour/common_vsilo_functionality.py

```diff
@@ -0,0 +1,677 @@
+    global flavour_params
+
+    # lets import the silo-controller-specific functions and bind them into us
+    print("importing module: " + broker_specific_module_name)
+    brokerspecific = import_module(broker_specific_module_name)
+
+    signal.signal(signal.SIGINT, handler)
+
+    # connect to the on-disk hashmap
+    shutil.rmtree('plyvelhashmap', ignore_errors=True)
+    leveldb = plyvel.DB('plyvelhashmap', create_if_missing=True)
+
+    # If you run this silo controller MANUALLY for DEBUG PURPOSES,
+    # then you may access the -d parameter to set a local address of the mongo DB
+    # and a local MQTT. This parameter is never visible when this silo
+    # is run via docker automatism (or similar) via the master controller.
+    parser = argparse.ArgumentParser(description='A vSilo Controller.')
+    parser.add_argument('-d', '--debug_mongo_ip', dest='debug_mongo_ip', help='run in debug mode and use the given IP address of the system MongoDB (usually 172.17.0.2)')
+    args = parser.parse_args()
+    # flavour_params must be a string
+    flavour_params = ""
+    if args.debug_mongo_ip != None:
+        # -------------- DEBUG PARAMETERS --------------
+        #tenant_id = "tenant1"
+        #flavour_params = ""
+        #v_silo_id = "tenant1_Silo1"
+        tenant_id = os.environ["tenantID"]
+        v_silo_id = os.environ["vSiloID"]
+        flavour_params = os.environ["flavourParams"]
+        virIoT_mqtt_data_broker_IP = "127.0.0.1"
+        virIoT_mqtt_data_broker_port = 1883
+        virIoT_mqtt_control_broker_IP = "127.0.0.1"
+        virIoT_mqtt_control_broker_port = 1883
+        db_IP = args.debug_mongo_ip
+        db_port = 27017
+        print("starting silo controller in DEBUG mode to " + db_IP)
+    else:
+        MAX_RETRY = 3
+        v_silo_id = os.environ["vSiloID"]
+        db_IP = os.environ['systemDatabaseIP']  # IP address of system database
+        db_port = os.environ['systemDatabasePort']  # port of system database
+
+        # Mongodb settings
+        time.sleep(1.5)  # wait before query the system database
+        db_name = "viriotDB"  # name of system database
+        v_thing_collection = "vThingC"
+        v_silo_collection = "vSiloC"
+        db_client = MongoClient('mongodb://' + db_IP + ':' + str(db_port) + '/')
+        db = db_client[db_name]
+        silo_entry = db[v_silo_collection].find_one({"vSiloID": v_silo_id})
+
+        valid_silo_entry = False
+        for x in range(MAX_RETRY):
+            if silo_entry is not None:
+                valid_silo_entry = True
+                break
+            time.sleep(3)
+
+        if not valid_silo_entry:
+            print("Error: Virtual Silo entry not found for v_silo_id:", v_silo_id)
+            exit()
+
+        try:
+            # import paramenters from DB
+            tenant_id = silo_entry["tenantID"]
+            flavourParams = silo_entry["flavourParams"]  # in this flavour, param is the silo type (Raw, Mobius, FiWare)
```

Comment:
I think maybe the variable flavourParams should be renamed as flavour_params, to make it work not in the debug mode.

Reply:
changing flavourParams to flavour_params fixed the start of the silo controller.
The change was added [here](https://github.com/fed4iot/VirIoT/pull/22/commits/f5ff29718109394289bcb2ab56699c14e2e94c7d#diff-a20868b0cbf3bab921a2905d873229dcR584)